src/vs/workbench/parts/files/common/explorerModel.ts

```diff
@@ -181,64 +182,63 @@ export class FileStat implements IFileStat {
 		if (mergingDirectories && disk.isDirectoryResolved) {
 
 			// Map resource => stat
-			const oldLocalChildren = new ResourceMap<FileStat>();
+			const oldLocalChildren = new ResourceMap<ExplorerItem>();
 			if (local.children) {
-				local.children.forEach((localChild: FileStat) => {
-					oldLocalChildren.set(localChild.resource, localChild);
-				});
+				for (let name in local.children) {
+					oldLocalChildren.set(local.children[name].resource, local.children[name]);
```

Comment:
@isidorn suggest to extract a local variable to prevent duplicate access (`var foo = local.children[name]`)